### PR TITLE
Protocol 1.8: fix duplicated vector component in spawn packet

### LINKED ISSUE #5306: Entities positions are not sent immediately
Client version: 1.8.9
Server OS: Linux Debian x64
Cuberite Commit id: 3b47eaadb85ac42b70d35d03d919181830b40a41

### Expected behavior
Entities appear where they are placed/spawned

### Actual behavior
Entities seem to interpolate from a different position to their spawnpoint. I was merging the armor stand patch and it seems to be lerping from z64

Doesn't seem to affect spawn eggs with monsters or mobs

### Steps to reproduce the behavior
Use 1.8.9 (doesn't seem noticable in 1.12) and place a boat or other entity you can place with a position.

### PR DESCRIPTION
**Pull request by tigerw**
* Fixes #5306

### CODE REVIEW

File: src/Protocol/Protocol_1_8.cpp [hunk 1]
```diff
@@ -2997,27 +2997,22 @@ void cProtocol_1_8_0::SendEntitySpawn(const cEntity & a_Entity, const UInt8 a_Ob
...
+	Pkt.WriteBEInt32(a_ObjectData);
+
+	if (a_ObjectData != 0)
 	{
-		cPacketizer Pkt(*this, pktSpawnObject);
-		Pkt.WriteVarInt32(a_Entity.GetUniqueID());
-		Pkt.WriteBEUInt8(a_ObjectType);
-		Pkt.WriteFPInt(a_Entity.GetPosX());  // Position appears to be ignored...
-		Pkt.WriteFPInt(a_Entity.GetPosY());
-		Pkt.WriteFPInt(a_Entity.GetPosY());
```
Comment (tigerw) [inline src/Protocol/Protocol_1_8.cpp:3006]: eks why _why_?

**Review by bearbin — COMMENTED** [review]
Why have you removed the brackets around the packet sending? This will mess up the packet ordering - the sendEntityTeleport packet will get sent first, and the spawnobject after in your changed version.

**Comment by tigerw** [comment]
The teleport was to work around the bug a few lines above, now it's been removed.

**Review by bearbin — APPROVED** [review]
Ah, OK I didn't notice that the teleport had been deleted :facepalm:

**Pull request merged**